ip_list.go

```diff
@@ -0,0 +1,445 @@
+	if err != nil {
+		return IPList{}, errors.Wrap(err, errMakeRequestError)
+	}
+
+	result := IPListResponse{}
+	if err := json.Unmarshal(res, &result); err != nil {
+		return IPList{}, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return result.Result, nil
+}
+
+// DeleteIPList deletes an IP List
+// API reference: https://api.cloudflare.com/#rules-lists-delete-list
+func (api *API) DeleteIPList(ctx context.Context, id string) (IPListDeleteResponse, error) {
+	uri := fmt.Sprintf("/accounts/%s/rules/lists/%s", api.AccountID, id)
+	res, err := api.makeRequestContext(ctx, http.MethodDelete, uri, nil)
+	if err != nil {
+		return IPListDeleteResponse{}, errors.Wrap(err, errMakeRequestError)
+	}
+
+	result := IPListDeleteResponse{}
+	if err := json.Unmarshal(res, &result); err != nil {
+		return IPListDeleteResponse{}, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return result, nil
+}
+
+// ListIPListItems returns a list with all items in an IP List
+// API reference: https://api.cloudflare.com/#rules-lists-list-list-items
+func (api *API) ListIPListItems(ctx context.Context, id string) ([]IPListItem, error) {
+	var list []IPListItem
+	var cursor string
+	var cursorQuery string
+
+	for {
+		if len(cursor) > 0 {
+			cursorQuery = fmt.Sprintf("?cursor=%s", cursor)
+		}
+		uri := fmt.Sprintf("/accounts/%s/rules/lists/%s/items%s", api.AccountID, id, cursorQuery)
+		res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+		if err != nil {
+			return []IPListItem{}, errors.Wrap(err, errMakeRequestError)
+		}
+
+		result := IPListItemsListResponse{}
+		if err := json.Unmarshal(res, &result); err != nil {
+			return []IPListItem{}, errors.Wrap(err, errUnmarshalError)
+		}
+
+		list = append(list, result.Result...)
+		if cursor = result.ResultInfo.Cursors.After; cursor == "" {
+			break
+		}
+	}
+
+	return list, nil
+}
+
+// CreateIPListItemAsync creates a new IP List Item asynchronously, Users have to poll the operation status by
+// using the operation_id returned by this function.
+// API reference: https://api.cloudflare.com/#rules-lists-create-list-items
+func (api *API) CreateIPListItemAsync(ctx context.Context, id string, ip string, comment string) (
+	IPListItemCreateResponse, error) {
+	uri := fmt.Sprintf("/accounts/%s/rules/lists/%s/items", api.AccountID, id)
+	res, err := api.makeRequestContext(ctx, http.MethodPost, uri, []IPListItemCreateRequest{{IP: ip, Comment: comment}})
+	if err != nil {
+		return IPListItemCreateResponse{}, errors.Wrap(err, errMakeRequestError)
+	}
+
+	result := IPListItemCreateResponse{}
+	if err := json.Unmarshal(res, &result); err != nil {
+		return IPListItemCreateResponse{}, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return result, nil
+}
+
+// CreateIPListItem creates a new IP List Item synchronously and returns the current set of IP List Items
+func (api *API) CreateIPListItem(ctx context.Context, id string, ip string, comment string) ([]IPListItem, error) {
```

Comment:
Similar to the suggestion above, can we omit the extra types in this method signature since they are duplicated?

```suggestion
func (api *API) CreateIPListItem(ctx context.Context, id, ip, comment string) ([]IPListItem, error) {
```